src/solver/infeasible-problem-analysis/watched-constraints.cpp

```diff
@@ -0,0 +1,238 @@
+
+#include "antares/solver/infeasible-problem-analysis/watched-constraints.h"
+
+#include <boost/algorithm/string.hpp>
+#include <boost/algorithm/string/regex.hpp>
+#include <boost/regex.hpp>
+
+class StringIsNotWellFormated: public std::runtime_error
+{
+public:
+    StringIsNotWellFormated(const std::string& error_message):
+        std::runtime_error(error_message)
+    {
+    }
+};
+
+std::string StringBetweenAngleBrackets(const std::string& constraintName)
+{
+    std::vector<std::string> split_name;
+    boost::split(split_name, constraintName, boost::is_any_of("<>"));
+
+    std::string err_msg = "Error: ";
+    if (split_name.size() < 3)
+    {
+        err_msg += "constraint name '" + constraintName + "' misses '<' and/or '>' bracket";
+        throw StringIsNotWellFormated(err_msg);
+    }
+    if (split_name[1].empty())
+    {
+        err_msg += "constraint name '" + constraintName + "' must be of format '*<str>*'";
+        throw StringIsNotWellFormated(err_msg);
+    }
+    return split_name[1];
+}
+
+std::string timeStep(std::vector<std::string> splitName)
+{
+    return StringBetweenAngleBrackets(splitName.at(splitName.size() - 2));
+}
+
+std::string shortName(std::vector<std::string> splitName)
+{
+    return splitName.at(0);
+}
+
+std::string areaName(std::vector<std::string> splitName)
+{
+    return StringBetweenAngleBrackets(splitName.at(1));
+}
+
+std::string STSname(std::vector<std::string> splitName)
+{
+    return StringBetweenAngleBrackets(splitName.at(2));
+}
+
+namespace Antares::Optimization
+{
+
+// =======================================
+// Generic constraint logger
+// =======================================
+void WatchedConstraint::setConstraintName(std::string name)
```

Comment:
```suggestion
void WatchedConstraint::setConstraintName(const std::string& name)
```

Reply:
or better set the name in the ctor?

Reply:
**setConstraintName** was removed, and replaced with a constructor, as you said.